src/sentry/grouping/variants.py

```diff
@@ -22,13 +23,14 @@ class FingerprintVariantMetadata(TypedDict):
     matched_rule: NotRequired[str]
 
 
-class BaseVariant:
-    # The type of the variant that is reported to the UI.
-    type: str | None = None
-
+class BaseVariant(ABC):
     # This is true if `get_hash` does not return `None`.
     contributes = True
 
+    @property
+    @abstractmethod
+    def type(self) -> str: ...
```

Comment:
this doesn't actually work -- an abstract instance property cannot be accessed on the class

it's probably best to just do `type: ClassVar[str]  # abstract` instead